src/slam_karto.cpp

```diff
@@ -291,12 +345,19 @@ SlamKarto::~SlamKarto()
     transform_thread_->join();
     delete transform_thread_;
   }
+  if(vis_thread_)
+  {
+    vis_thread_->join();
+    delete vis_thread_;
+  }
+  if(solver_)
```

Comment:
ok @stwirth, found the problem. `GraphVisualizer` is abstract and it was missing a virtual destructor, so the destructor of `SPAGraphVisualizer` was never called, and the references to the smart pointers were never decremented.
Now fixed, you can check with `roslaunch slam_karto karto_slam.launch`